### DISCUSSION TITLE
能否增加熔断插件对多节点的支持

### DISCUSSION THREAD

**Discussion by zhangyial** [apache/apisix#7341] [Q&A]
目前apisix通过维护请求的非健康|健康的计数来控制熔断逻辑。
假设我们有2个节点A和B，设置的不健康状态的连续错误请求次数为3
由于非健康计数器以内存方式保存在节点本地，可能节点A连续3次不健康请求已经触发了熔断、节点B还未触发
然而有时业务对熔断要求比较敏感
是否可以将熔断计数变量保存在中心，A，B节点共享变量，这样所有的节点的熔断状态就能一致了

**Comment by zhangyial** [comment]
我尝试基于现有逻辑改造了api-breaker插件，想将将计数保存在redis里，但是熔断插件对请求结果的计数是在log阶段，由于log_by_lua对cosocket函数类的限制，导致在log阶段不能使用resty.redis api，请问实现熔断插件对多节点支持是否有其他思路
![image](https://user-images.githubusercontent.com/32232794/176085570-612ff3a8-4592-434c-aaed-70ab377ed96e.png)

**Reply by tzssangglass** [reply]
We can use a timer to connect to redis, but note that this is asynchronous, so it may not be as accurate in strict scenarios.
